src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -117,40 +247,165 @@ public SimpleQueryStringBuilder flags(SimpleQueryStringFlag... flags) {
+    }
+
+    /** Set the name for this query. */
+    public SimpleQueryStringBuilder queryName(String queryName) {
+        this.queryName = queryName;
+        return this;
+    }
+
+    /** Returns the name for this query. */
+    public String queryName() {
+        return queryName;
+    }
+
+    /**
+     * Specifies whether parsed terms for this query should be lower-cased.
+     * Defaults to true if not set.
+     */
+    public SimpleQueryStringBuilder lowercaseExpandedTerms(Boolean lowercaseExpandedTerms) {
+        this.settings.lowercaseExpandedTerms(lowercaseExpandedTerms);
         return this;
     }
 
+    /** Returns whether parsed terms should be lower cased for this query. */
+    public Boolean lowercaseExpandedTerms() {
+        return this.settings.lowercaseExpandedTerms;
+    }
+
+    /** Specifies the locale for parsing terms. Defaults to ROOT if none is set. */
     public SimpleQueryStringBuilder locale(Locale locale) {
-        this.locale = locale;
+        this.settings.locale(locale);
         return this;
     }
 
-    public SimpleQueryStringBuilder lenient(boolean lenient) {
-        this.lenient = lenient;
+    /** Returns the locale for parsing terms for this query. */
+    public Locale locale() {
+        return this.settings.locale;
+    }
+
+    /** Specifies whether query parsing should be lenient. Defaults to false. */
+    public SimpleQueryStringBuilder lenient(Boolean lenient) {
+        this.settings.lenient(lenient);
         return this;
     }
 
-    public SimpleQueryStringBuilder analyzeWildcard(boolean analyzeWildcard) {
-        this.analyzeWildcard = analyzeWildcard;
+    /** Returns whether query parsing should be lenient. */
+    public Boolean lenient() {
+        return this.settings.lenient();
+    }
+
+    /** Specifies whether wildcards should be analyzed. Defaults to false. */
+    public SimpleQueryStringBuilder analyzeWildcard(Boolean analyzeWildcard) {
+        this.settings.analyzeWildcard(analyzeWildcard);
         return this;
     }
 
+    /** Returns whether wildcards should by analyzed. */
+    public Boolean analyzeWildcard() {
+        return this.settings.analyzeWildcard();
+    }
+
+    /**
+     * Specifies the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
     public SimpleQueryStringBuilder minimumShouldMatch(String minimumShouldMatch) {
         this.minimumShouldMatch = minimumShouldMatch;
         return this;
     }
 
+    /**
+     * Returns the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
+    public String minimumShouldMatch() {
+        return minimumShouldMatch;
+    }
+
+    /** Returns an exception if any mandatory fields are null. */
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+
+        // Query text is required
+        if (queryText == null) {
```

Comment:
If I'm reading the oal.queryparser.simple.SimpleQueryParser:153 correctly it's perfectly happy parsing an empty string and returns http://lucene.apache.org/core/5_1_0/core/org/apache/lucene/search/MatchNoDocsQuery.html

Adjusted the unit test to also generate empty query strings and the test doesn't crash but returns the above query object as expected.